File: Flowsource-templates/springboot3-azure-ci-pipeline-template/skeleton/src/test/java/${{values.java_package_name_slash}}/exception/ResourceNotFoundExceptionTest.java
package ${{values.java_package_name}}.exception;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

public class ResourceNotFoundExceptionTest {
    @Test
    public void testConstructorWithMessage() {
        String message = "Resource not found";
        ResourceNotFoundException exception = new ResourceNotFoundException(message);

        assertEquals(message, exception.getMessage());
    }
    
    @Test
    public void testConstructorWithMessageAndCause() {
        String message = "Resource not found";
        Throwable cause = new RuntimeException("Internal error");

        ResourceNotFoundException exception = new ResourceNotFoundException(message, cause);

        assertEquals(message, exception.getMessage());
        assertNotNull(exception.getCause());
        assertEquals("Internal error", exception.getCause().getMessage());
    }

    @Test
    public void testConstructorWithInvalidId() {
        Long invalidId = 123L;
        ResourceNotFoundException exception = new ResourceNotFoundException(invalidId);

        assertEquals("Resource with id " + invalidId + " not found", exception.getMessage());
    }

}

File: Flowsource-templates/python-github-aws-ci-pipeline-template/skeleton/app/test/exception/resource_not_found_exception_test.py
from app.exception.resource_not_found_exception import ResourceNotFoundException

def test_constructor_with_message():

    message = "Resource not found"
    exception = ResourceNotFoundException(message)
    assert exception.args[0] == message

def test_constructor_with_message_and_cause():

    message = "Resource not found"
    cause = Exception("Internal error")
    exception = ResourceNotFoundException(message, cause=cause)
    assert exception.args[0] == message
    assert isinstance(exception.__cause__, type(cause))
    assert str(exception.__cause__) == "Internal error"

def test_from_invalid_id():

    invalid_id = 123
    exception = ResourceNotFoundException.from_invalid_id(invalid_id)
    assert exception.args[0] == f"Resource with id {invalid_id} not found"
